src/v8n.js

```diff
@@ -40,19 +40,16 @@ function proxyContext(context) {
 }
 
 function proxyContextEs5(context) {
-  Object.keys(availableRules).forEach(prop => {
-    context[prop] = (...args) => {
-      const newContext = proxyContextEs5(context._clone());
-      return newContext._applyRule(availableRules[prop], prop)(...args);
-    };
-  });
+  const buildRuleSet = ruleSet =>
```

Comment:
To avoid side effects this should accept the context as a parameter and return it. This is currently modifying a variable in place and is not very readable.

Also: Does this name make sense? This adds a rule set to the context and not only builds it. Just as a side note.